proto/src/lib.rs

```diff
@@ -77,7 +82,16 @@ pub mod sighash {
                 Some(TxAction::Output(o)) => Some(SHAction::Output(o)),
                 Some(TxAction::Delegate(d)) => Some(SHAction::Delegate(d)),
                 Some(TxAction::Undelegate(d)) => Some(SHAction::Undelegate(d)),
-                Some(TxAction::ValidatorDefinition(d)) => Some(SHAction::ValidatorDefinition(d)),
+                // The `ValidatorDefinition` contains sig bytes, so we only want to hash the `Validator` itself.
+                // Some(TxAction::ValidatorDefinition(v)) => Some(SHAction::Validator(v)),
```

Comment:
Either way would be fine here, but also, we should do a proper sighash definition later, so that we don't need to do ad-hoc proto stuff for signing.

Reply:
How do you mean? Is there an example in the codebase?

Reply:
> In this specific case, it's fine to have the full validator definition be part of the sighash data, because the validator definition signature is part of the transaction data to be signed, rather than itself being a spend authorization signature (the validator definition signature binds only to the definition, not to the whole transaction it's part of).

This is not actually true, right now the entire transaction is signed (and the verification checks that the validator's identity key made the signature), not just the validator definition itself

Reply:
Hmm, I think the signature in the `ValidatorDefinition` should only be signing the `Validator`, since otherwise we have no way to verify the signatures later: https://github.com/penumbra-zone/penumbra/pull/480/files#r827261810